src/tbb/scheduler_common.h

```diff
@@ -439,6 +440,33 @@ struct suspend_point_type {
 #pragma warning( disable: 4324 )
 #endif
 
+class thread_reference_vertex: public d1::reference_vertex {
+public:
+    using d1::reference_vertex::reference_vertex;
+
+    void reserve(std::uint32_t delta = 1) override {
+        if (reserve_impl(delta) == 0) {
+            m_lifetime_ref_counter.fetch_add(1);
+        }
+    }
+
+    void release(std::uint32_t delta = 1) override {
+        if (release_impl(delta) == 0) {
+            destroy();
+        }
+    }
+
+    void destroy() {
```

Comment:
I would suggest alternative naming for this method, since it does not actually destroy the vertex any time you call it. 
May be something like `release_ownership`?